federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -393,41 +393,6 @@ func (s *ServiceController) updateFederationService(key string, cachedService *c
 	return nil, !retryable
 }
 
-func (s *ServiceController) deleteFederationService(cachedService *cachedService) (error, bool) {
-	// handle available clusters one by one
-	var hasErr bool
-	for clusterName, cluster := range s.clusterCache.clientMap {
-		err := s.deleteClusterService(clusterName, cachedService, cluster.clientset)
-		if err != nil {
-			hasErr = true
-		} else if err := s.ensureDnsRecords(clusterName, cachedService); err != nil {
```

Comment:
I have a PR for it: https://github.com/kubernetes/kubernetes/pull/36390.
Also tracked in https://github.com/kubernetes/kubernetes/issues/33612.